packages/features/calendars/DatePicker.tsx

```diff
@@ -192,10 +192,15 @@ const Days = ({
       // If selected date not available in the month, select the first available date of the month
       props.onChange(firstAvailableDateOfTheMonth);
     }
-
+    if (isSelectedDateAvailable) {
+      props.onChange(dayjs(selected));
+    }
     if (!firstAvailableDateOfTheMonth) {
       props.onChange(null);
     }
+    return () => {
+      props.onChange(null);
+    };
```

Comment:
what do these do?

Reply:
- After we open an event from the user page and then click back button the date queryParam is removed but the selectedDate is still present in the store. So this code sets the Date queryParam on subsequent visit to event page.
- The cleanup function makes sure to remove the query parameter on unmounting ( going back to user page)

Reply:
ahh I see. @CarinaWolli @emrysal whats your take on this?